src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.GuardedCallsTests.cs

```diff
@@ -1216,6 +1217,76 @@ static void M1()
             await VerifyAnalyzerCSAsync(csSource);
         }
 
+        [Fact]
+        public async Task CallGuardFromPlatformSpecificAssembly()
+        {
+            string csDependencyCode = @"
+public class Library
+{
+    [System.Runtime.Versioning.SupportedOSPlatformGuard(""windows"")]
+    [System.Runtime.Versioning.SupportedOSPlatformGuard(""linux"")]
+    public static bool IsSupported => false;
+
+    public static void AMethod() { }
+}";
+            csDependencyCode = @$"[assembly: System.Runtime.Versioning.SupportedOSPlatform(""windows"")]
+                                  [assembly: System.Runtime.Versioning.SupportedOSPlatform(""linux"")]
+                                  {csDependencyCode}";
+
+            string csCurrentAssemblyCode = @"
+using System;
+
+public class Program
+{
+    public void ProgramMethod()
+    {
+        [|Library.AMethod()|]; // Not guarded, warns
+        if (Library.IsSupported)
+        {
+             Library.AMethod();
+        }
+    }
+}";
+            var test = SetupDependencyAndTestCSWithOneSourceFile(csCurrentAssemblyCode, csDependencyCode);
+            test.TestState.AnalyzerConfigFiles.Add(("/.editorconfig", $@"root = true
+
+[*]
+build_property.TargetFramework = net5
```

Comment:
Would it make sense to make this test method a Theory and pass all the .NET versions that support this? Or is it not worth testing the code in all the .NET versions equal or greater than .NET 5?

Reply:
So, the analyzer bails out when project TargetFramework is below .NET 5.0 or missing, that is why it is added here, there are existing tests that test this scenario